File: 프로그래머스/0/120850. 문자열 정렬하기 （1）/문자열 정렬하기 （1）.js
function solution(my_string) {
    // 영문자 제외 정규 표현식, 치환된 공백 제거
    var pattern = my_string.replace(/[A-Z]/ig, '').split('');
    var answer = [];
    
    // 문자열 배열 => 숫자열 배열 변환, 정렬
    return answer = pattern.map(Number).sort();
}

File: 프로그래머스/0/120850. 문자열 정렬하기 （1）/문자열 정렬하기 （1）.java
import java.util.Arrays;
import java.util.regex.Pattern;
import java.util.stream.Stream;

class Solution {
    public int[] solution(String my_string) {
        // 정규 표현식 패턴 컴파일
        Pattern pattern = Pattern.compile("[a-zA-Z]+");
        
        // String.split("") => 문자열을 한 글자씩 나눔
        // 나눈 문자를 정규 표현식과 일치하는지 확인
        // 패턴과 일치하지 않는 문자들만 필터링해서 배열 생성
        String[] str = Stream.of(my_string.split(""))
            .filter(s -> !pattern.matcher(s).matches())
            .toArray(String[]::new);
        
        int[] answer = new int[str.length];
        
        // 문자열 배열 => 숫자 배열 변환
        for (int i = 0; i < str.length; i++) {
            answer[i] = Integer.parseInt(str[i]);
        }
        
        Arrays.sort(answer); // 오름차순 정렬
        
        return answer;
    }
}